File: Examples/CSharp/Mail-Merge/ProduceMultipleDocuments.cs
using Aspose.Words;
using Aspose.Words.MailMerging;
using System;
using System.Collections.Generic;
using System.Data;
using System.Data.OleDb;
using System.Diagnostics;
using System.IO;
using System.Linq;
using System.Text;
using System.Web;

namespace Aspose.Words.Examples.CSharp.Mail_Merge
{
    class ProduceMultipleDocuments
    {
        public static void Run()
        {
            //ExStart:ProduceMultipleDocuments            
            //Put the path to the documents directory and open the template:
            string dataDir = RunExamples.GetDataDir_MailMergeAndReporting();
            Document doc = new Document(dataDir + "TestFile.doc");

            // Open the database connection.
            string connString = @"Provider=Microsoft.Jet.OLEDB.4.0;Data Source=" + dataDir + "Customers.mdb";
            OleDbConnection conn = new OleDbConnection(connString);
            conn.Open();

            // Get data from a database.
            OleDbCommand cmd = new OleDbCommand("SELECT * FROM Customers", conn);
            OleDbDataAdapter da = new OleDbDataAdapter(cmd);
            DataTable data = new DataTable();
            da.Fill(data);

            //Perform a loop through each DataRow to iterate through the DataTable.
            //Clone the template document instead of loading it from disk for better speed performance before the mail merge operation.
            //You can load the template document from a file or stream but it is faster to load the document only once and then clone it in memory before each mail merge operation.

            int counter = 1;
            foreach (DataRow row in data.Rows)
            {
                Document dstDoc = (Document) doc.Clone(true);
                dstDoc.MailMerge.Execute(row);
                dstDoc.Save(string.Format(dataDir + "TestFile_out{0}.doc", counter++));
            }

            Console.WriteLine("\nProduce multiple documents performed successfully.\nFile saved at " + dataDir);
            //ExEnd:ProduceMultipleDocuments           
        }
    }
}

File: Examples/VisualBasic/Mail-Merge/ProduceMultipleDocuments.vb
Imports Aspose.Words
Imports Aspose.Words.MailMerging
Imports System.Collections.Generic
Imports System.Data
Imports System.Data.OleDb
Imports System.Diagnostics
Imports System.IO
Imports System.Linq
Imports System.Text
Imports System.Web
Public Class ProduceMultipleDocuments
    Public Shared Sub Run()
        ' ExStart:ProduceMultipleDocuments            
        ' The path to the documents directory.
        Dim dataDir As String = RunExamples.GetDataDir_MailMergeAndReporting()
        ' Open the database connection.
        Dim connString As String = (Convert.ToString("Provider=Microsoft.Jet.OLEDB.4.0;Data Source=") & dataDir) + "Customers.mdb"
        Dim conn As New OleDbConnection(connString)
        conn.Open()
        ' Get data from a database.
        Dim cmd As New OleDbCommand("SELECT * FROM Customers", conn)
        Dim da As New OleDbDataAdapter(cmd)
        Dim data As New DataTable()
        da.Fill(data)

        ' Open the template document.
        Dim doc As New Document(dataDir & Convert.ToString("TestFile.doc"))

        Dim counter As Integer = 1
        ' Loop though all records in the data source.
        For Each row As DataRow In data.Rows
            ' Clone the template instead of loading it from disk (for speed).
            Dim dstDoc As Document = DirectCast(doc.Clone(True), Document)

            ' Execute mail merge.
            dstDoc.MailMerge.Execute(row)

            ' Save the document.
            dstDoc.Save(String.Format(dataDir & Convert.ToString("TestFile_out{0}.doc"), System.Math.Max(System.Threading.Interlocked.Increment(counter), counter - 1)))
        Next
        ' ExEnd:ProduceMultipleDocuments
        Console.WriteLine(Convert.ToString(vbLf & "Produce multiple documents performed successfully." & vbLf & "File saved at ") & dataDir)
    End Sub
End Class
